File: booking-service/src/main/legacy/java/entelect/training/incubator/spring/booking/legacy/error/GlobalErrorHandler.java
package entelect.training.incubator.spring.booking.legacy.error;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseStatus;
import org.springframework.web.bind.annotation.RestControllerAdvice;

import java.util.Map;

@RestControllerAdvice
public class GlobalErrorHandler {

    @ExceptionHandler(CustomerNotFoundException.class)
    @ResponseStatus(HttpStatus.NOT_FOUND)
    public Map<String, String> handleCustomerNotFound(CustomerNotFoundException ex) {
        return Map.of(
                "error", "CUSTOMER_NOT_FOUND",
                "message", ex.getMessage()
        );
    }

}

File: booking-service/src/main/kotlin/entelect/training/incubator/spring/booking/error/GlobalErrorHandler.kt
package entelect.training.incubator.spring.booking.error

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestControllerAdvice
import java.util.Map


@RestControllerAdvice
class GlobalErrorHandler {
    @ExceptionHandler(CustomerNotFoundException::class)
    @ResponseStatus(HttpStatus.NOT_FOUND)
    fun handleCustomerNotFound(ex: CustomerNotFoundException): MutableMap<String?, String?> {
        return Map.of<String?, String?>(
            "error", "CUSTOMER_NOT_FOUND",
            "message", ex.message
        )
    }
}
